Index: src/test/java/StateTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/StateTest.java b/src/test/java/StateTest.java
new file mode 100644
--- /dev/null	(date 1686675618668)
+++ b/src/test/java/StateTest.java	(date 1686675618668)
@@ -0,0 +1,29 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StateTest {
+    Tree<Integer, Pessoa> treeCPF = new Tree<>();
+    State stateTeste = new State();
+
+    Pessoa joao = new Pessoa(155465, 1231243, 2004, 4, 4, "joão", "jãolandia");
+    Pessoa maria = new Pessoa(104654654, 465465, 2004, 5, 4, "Maria", "Sapucaia do Norte");
+    Pessoa caludete = new Pessoa(7895456, 45612465, 2025, 5, 13, "Caludete", "Uberlândia");
+    Pessoa joaquim = new Pessoa(92342334, 54765734, 2004, 2, 29, "joaquim", "joaquimlandia");
+
+    @BeforeEach
+    void arvorePreMontada() {
+        stateTeste.adicionaPessoa(joao);
+        stateTeste.adicionaPessoa(maria);
+        stateTeste.adicionaPessoa(caludete);
+        stateTeste.adicionaPessoa(joaquim);
+
+    }
+
+    @Test
+    void pesquisaPorCPF() {
+        Pessoa p = stateTeste.pesquisaPorCPF(7895456);
+        assertEquals(caludete, p);
+    }
+}
\ No newline at end of file
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n<project version=\"4\">\r\n  <component name=\"ExternalStorageConfigurationManager\" enabled=\"true\" />\r\n  <component name=\"MavenProjectsManager\">\r\n    <option name=\"originalFiles\">\r\n      <list>\r\n        <option value=\"$PROJECT_DIR$/pom.xml\" />\r\n      </list>\r\n    </option>\r\n  </component>\r\n  <component name=\"ProjectRootManager\" version=\"2\" languageLevel=\"JDK_17\" default=\"true\" project-jdk-name=\"17\" project-jdk-type=\"JavaSDK\">\r\n    <output url=\"file://$PROJECT_DIR$/out\" />\r\n  </component>\r\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/misc.xml b/.idea/misc.xml
--- a/.idea/misc.xml	(revision 79c80958f43960c57bdb873df807502c67a2eb5a)
+++ b/.idea/misc.xml	(date 1686675095761)
@@ -8,7 +8,7 @@
       </list>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_17" default="true" project-jdk-name="17" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_20" default="true" project-jdk-name="openjdk-20" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/out" />
   </component>
 </project>
\ No newline at end of file
